tuist/tuist
# Add support for disabling `Mac (designed for iOS)` destination for iOS deployment target

### Short description 📝

As discussed in `https://tuistapp.slack.com/archives/CCXC295QS/p1678042685997789` currently when the deployment target is set to something similar to `.iOS(targetVersion: "16.0", devices: .iphone)`,
- The generated `.plist` contains values set for `UISupportedInterfaceOrientations~ipad`
- There is no way to omit support for Mac (designed for iPhone)

> Describe here the purpose of your PR.

The purpose of this PR is to have the deployment target better reflected in the generated `.plist` and give the users the option to opt out of `mac (designed for iPhone)` option.

### How to test the changes locally 🧐

In a tuist-driven project: 
- modify the `deploymentTarget` of a target to include only `.iphone` for devices. The `.plist` of the generated project's target should not contain `UISupportedInterfaceOrientations~ipad`.
- add `supportsMacDesignedForIPhone` set to `false` in the `deploymentTarget` (defaults to true) and the generated target should not have `Mac (designed for iPhone)` under the Supported Destinations section.

### Contributor checklist ✅


### Reviewer checklist ✅
